### PR TITLE
This is NoParameters, my latest contribution to the problem of an ind…

### PR DESCRIPTION
**Pull request by FranklinTOliveira**
…ex page for the root directory. NoParameters is a Git where I adjust the template for the webpage body to be operate without parameters in the Java constructor.

### CODE REVIEW

**Comment by FranklinTOliveira** [comment]
I've addressed the problem of adjusting the template for HomePage and NoParameters is the Git where we review the basic, initial work. But it will require finetuning and thus future Gits/Pull Requests to go over my future work where I do stuff like introduce the Bootstrap CSS and address what happens to null and empty strings after localhost as opposed to after /houses/

I also want the Frontispiece introPanel to be more like its namesake by introducing an image and a brief appropriate comment. But...

...that's all stuff for future Gits/Pull Requests. Right now, I'm just introducing the template to HomePage in a way that appears in the browser, however much it's a patchwork project at this stage!

File: src/main/java/website/fgto/royals_of_britain/view/HomePage.html [hunk 1]
```diff
@@ -1,63 +1,60 @@
...
+					</li>
+				</ul>
+			</div>
+
+			<div class="col-sm-9 position-relative d-flex align-items-end flex-column p-2">
+				<main role="main">
+					<div id="intro" class="row">
+						<p>Welcome to the beginning of our genealogical journey!</p>
+					</div>
+					<div id="table" class="row">
```
Comment (kyleoliveira) [inline src/main/java/website/fgto/royals_of_britain/view/HomePage.html:32]: Why would you have this table?

File: src/main/java/website/fgto/royals_of_britain/view/HomePage.java [hunk 2]
```diff
@@ -1,15 +1,44 @@
...
+  @Override
+  public void renderHead(IHeaderResponse response) {
+    CssResourceReference cssFile =
+        new CssResourceReference(WicketApplication.class,"style.css");
+    CssHeaderItem cssItem = CssHeaderItem.forReference(cssFile);
+
+    response.render(cssItem);
+  }
+
+  public void addIntroPanel(String houseName) {
```
Comment (kyleoliveira) [inline src/main/java/website/fgto/royals_of_britain/view/HomePage.java:36]: This method isn't getting called in your constructor, so it isn't getting used. If you're seeing Wicket exceptions, that's probably why...

File: src/main/java/website/fgto/royals_of_britain/view/houses/HouseInformationPage.java [hunk 3]
```diff
@@ -36,7 +36,7 @@ public void renderHead(IHeaderResponse response) {
     response.render(cssItem);
   }
 
-  private void addIntroPanel(String houseName) {
+  public void addIntroPanel(String houseName) {
```
Comment (kyleoliveira) [inline src/main/java/website/fgto/royals_of_britain/view/houses/HouseInformationPage.java:39]: This doesn't need to change.

File: src/main/java/website/fgto/royals_of_britain/view/HomePage.java (hunk 2 above)
Comment (FranklinTOliveira) [reply src/main/java/website/fgto/royals_of_britain/view/HomePage.java:36]: Yes, introPanel goes with the parameters I'm removing. Good point!

File: src/main/java/website/fgto/royals_of_britain/view/houses/HouseInformationPage.java (hunk 3 above)
Comment (FranklinTOliveira) [reply src/main/java/website/fgto/royals_of_britain/view/houses/HouseInformationPage.java:39]: Yes if the line is unnecessary, then it's ultimately moot whether it stays the same or is changed (unless I get straight to the point and delete it).

File: src/main/java/website/fgto/royals_of_britain/view/HomePage.html (hunk 1 above)
Comment (FranklinTOliveira) [reply src/main/java/website/fgto/royals_of_britain/view/HomePage.html:32]: But I have a template.

Perhaps there should be an exception to the template, regarding the table. But then I should probably devise a way to carry over the exception to /houses/ as well.

Comment (kyleoliveira) [reply src/main/java/website/fgto/royals_of_britain/view/HomePage.html:32]: Why would your homepage - which is just showing the message from the frontispiece - have a table displaying kings? Especially when you don't have a particular house to focus on? You're not going to show every single king on this page, are you?

Comment (FranklinTOliveira) [reply src/main/java/website/fgto/royals_of_britain/view/HomePage.html:32]: You misunderstand me--I'm thinking ahead.

I propose that if HomePage can be without a table, then HouseInformationPage (without further parameters) can be likewise.

But then that poses the question of _how_, given that the introPanels link to the HTML through the wickets attached to the table.

Comment (kyleoliveira) [reply src/main/java/website/fgto/royals_of_britain/view/HomePage.html:32]: Why would you want HouseInformationPage to _not_ have a table? Isn't HouseInformationPage supposed to be a page that lists the kings of a given house? Seems like you'd just make a different page class if you wanted some other page that didn't have a table on it.

Comment (FranklinTOliveira) [reply src/main/java/website/fgto/royals_of_britain/view/HomePage.html:32]: So...

...localhost   should have no table...

...localhost/houses/     should have no table...

...but localhost/houses/bruce  should have a table and likewise if bruce is replaced by cenelngabran, hanover, windsor, etc.

Do you get what I mean by that?

Comment (kyleoliveira) [reply src/main/java/website/fgto/royals_of_britain/view/HomePage.html:32]: Yes, that makes sense, except that I'd propose that you change `/houses/` to show the Apology panel without a table. You don't really need the frontispiece thing anymore once you've updated the homepage to have the equivalent information - if somebody tries to go to `/houses/` that's just another error state.

That said, I think you should make the change to the pages that have tables in a separate PR.

**Pull request merged**